### PR TITLE
Update question/program translation views for consistency and usability

### PR DESCRIPTION
**Pull request by clouser2**
### Description

As part of adding translatable content for status tracking, we discussed improving the usability of the current translation views. Note that in a follow-on PR we'll be removing the English translation tab since it's duplicative with the existing content in the program/question configuration screens.

The UI changes here include (see screenshots below):
  * Adding the program / question name to the title so one knows which entity translations are being edited for
  * Add a consistent label for each field for what's being set. Previously, question translations used the English text for the field label
  * Using fieldsets for better information organization
  * Add a small section below the field that indicates the English version of the text (hidden for the English translation tab)

While updating, a few cleanups were made on the server side:
  * Marked a few classes as final to make the choice to subclass explicit
  * Switched browser test code to drive updates via user-visible text rather than by IDs. Allowed us to remove a few hard-coded IDs within the translations view.
 
## Screenshots

Questions (before):
Screen Shot 2022-08-04 at 9 47 32 AM

Questions (after):
Screen Shot 2022-08-04 at 9 18 31 AM

Programs (before):
Screen Shot 2022-08-04 at 9 47 15 AM

Programs (after):
Screen Shot 2022-08-04 at 9 19 12 AM

## Release notes:

Reorganizes program and question translation editing for increased usability.

### Checklist

- [X] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [X] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
#2754

### CODE REVIEW

File: server/app/views/admin/TranslationFormView.java [hunk 1]
```diff
@@ -93,4 +100,28 @@ protected <T extends Tag<T>> FormTag renderTranslationForm(
...
+  protected final FieldsetTag fieldSetForFields(
+      String legendText, ImmutableList<DomContent> fields) {
+    return fieldset()
+        .withClasses(Styles.MY_4, Styles.PT_1, Styles.PB_2, Styles.PX_2, Styles.BORDER)
+        .with(
+            legend(legendText), div().withClasses(Styles.FLEX_ROW, Styles.SPACE_Y_4).with(fields));
+  }
+
+  protected final boolean isDefaultLocale(Locale locale) {
+    // TODO(#2752): Remove isDefaultLocale once English translations have been disabled.
```
Comment (shanemc-goog) [inline server/app/views/admin/TranslationFormView.java:124]: This should be on/above the method not in it.

File: server/test/controllers/admin/AdminProgramTranslationsControllerTest.java [hunk 2]
```diff
@@ -94,8 +144,23 @@ public void update_validationErrors_rendersEditFormWithMessages() {
...
-            "Manage program translations",
+            String.format("Manage program translations: Internal program name"),
             "program display name cannot be blank",
             "program display description cannot be blank");
   }
+
+  private Program createProgram(
+      String adminName, LocalizedStrings localizedName, LocalizedStrings localizedDescription) {
+    Program initialProgram = ProgramBuilder.newDraftProgram(adminName).build();
+    // ProgamBuilder initializes the localized name and doesn't currently support providding
```
Comment (shanemc-goog) [inline server/test/controllers/admin/AdminProgramTranslationsControllerTest.java:155]:
```suggestion
    // ProgramBuilder initializes the localized name and doesn't currently support providing
```

**Review by shanemc-goog — APPROVED** [review]
just the nits

File: server/test/controllers/admin/AdminProgramTranslationsControllerTest.java (hunk 2 above)
Comment (clouser2) [reply server/test/controllers/admin/AdminProgramTranslationsControllerTest.java:155]: Done!

File: server/app/views/admin/TranslationFormView.java (hunk 1 above)
Comment (clouser2) [reply server/app/views/admin/TranslationFormView.java:124]: Done!

**Review by clouser2 — COMMENTED** [review]
Feedback incorporated.

**Review by shanemc-goog — APPROVED** [review]

**Pull request merged**